Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-Lists-LoadThenNavigate.swift

```diff
@@ -83,16 +65,22 @@ let loadThenNavigateListReducer =
       case let .setNavigationSelectionDelayCompleted(id):
         state.rows[id: id]?.isActivityIndicatorVisible = false
         state.selection = Identified(
-          CounterState(count: state.rows[id: id]?.count ?? 0),
+          Counter.State(count: state.rows[id: id]?.count ?? 0),
           id: id
         )
         return .none
       }
     }
-  )
+    .ifLet(\.selection, action: .self) {
+      Scope(state: \Identified<State.Row.ID, Counter.State>.value, action: /Action.counter) {
+        Counter()
+      }
+    }
```

Comment:
While migrating some code to the new reducer API, I was getting the runtime warning about receiving a child action when child state is nil in a situation where I didn't expect it. Can't guarantee it's not user error, but I was stumped. Went looking through the case studies and found this domain that matches mine: an optional `Identified` value driving list selection navigation. Ran the case study, and found that this composition here too emits the child action received when child state is nil runtime warning. Unsure if the warning is in error, or the sample code, or maybe this is a known thing addressed by the upcoming navigation changes. Feel free to close or disregard if this is known issue for the time being.

Reply:
@kamcma I think this might just be an order-of-operations thing with the nesting that we hope is solved by some of our navigation explorations. I haven't had time to run the case study, but do you have some steps to reproduce for when I dive in?

Reply:
I'm at the current head of the protocol-beta branch, e00022c. Using Xcode 14b6 on Apple silicon. Launch the SwiftUI case studies app in a simulator, navigate to the lists-load-then-navigate case study, tap the first counter in the list to initiate navigation to the optional counter starting at 1, note runtime warning.

Reply:
@kamcma That was actually a bug in my translation. I'd swapped the action case paths and used `.self` in the wrong spot! @mbrandonw fixed it the other day here: https://github.com/pointfreeco/swift-composable-architecture/pull/1283/commits/426d5c5932ba831b51cbc79701bbb9ea7891d0a2

Our navigation APIs should help avoid this problem since they will flatten things.